sqlglot/optimizer/simplify.py

```diff
@@ -276,7 +276,9 @@ def _simplify_comparison(expression, left, right, or_=False):
         rargs = {rl, rr}
 
         matching = largs & rargs
-        columns = {m for m in matching if isinstance(m, exp.Column)}
+        columns = {
+            m for m in matching if not isinstance(m, exp.Literal) and not _is_date_literal(m)
```

Comment:
I think this logic gets shaky on something like:

```sql
RANDOM() > 1 AND RANDOM() > 0
```

But even in that case, I think many engines will do common subexpression elimination and only calculate RANDOM once?

Reply:
Not sure this is a reasonable assumption, I think each `RANDOM()` call should yield different results. E.g.
```
georgesittas=# select random(), random();
       random        |      random
---------------------+-------------------
 0.44325693039061487 | 0.671197612400882
(1 row)
```

Reply:
why don't you just do is not _is_constant() and not random?

Reply:
Ah thanks for checking!

Well, you could have `1 + RANDOM()`, so we'd need to do something like `not _is_constant(m) and not m.find(exp.Random)`. And I'm worried that there might be other nondeterministic expressions we're over looking.

So maybe I'll start with something narrower for now...